### ISSUE TITLE
Inline 'std' directive not obeyed

### ISSUE THREAD

**Issue by mwild1** [mpeterv/luacheck#51]
Easiest with an example:

```
$ cat t.lua 
print("hello world")

$ luacheck -q --std none t.lua 
Checking t.lua                                    1 warning

    t.lua:2:1: accessing undefined variable print

Total: 1 warning / 0 errors in 1 file
```

Expected behaviour so far. But if I add the std none as an inline directive, it has no effect:

```
$ cat t.lua 
--luacheck: std none
print("hello world")

$ luacheck -q t.lua 
Total: 0 warnings / 0 errors in 1 file
```

**Comment by mpeterv** [comment]
Yes, right now inline options can be used only to remove some warnings, but not to re-add warnings removed by default. I'm not sure what's expected behaviour when `std` is used both inline and as an argument/in config. Probably sets of globals should be joined in that case.

**Comment by mwild1** [comment]
My expectation is that std has a set of values (none, min, max, etc.), and I expected std to be 'none' for this file, and override what was specified on the command line (or the default). Other files in the same run would be handled using the default std value.

If it helps, the root problem we are trying to solve is that luacheck does not detect:

```
_ENV = nil
print("hello world")
```

I expected that this would fix that:

```
--luacheck: std none
_ENV = nil
print("hello world")
```

**Comment by Zash** [comment]
The goal here is to point to a line and say "Any use of a global variable below this line is an error", such as with the common pattern of putting everything you are going to use into locals at the top of the file.

**Comment by mpeterv** [comment]
This should be fixed on master branch, sorry to keep you waiting. Can you test it?

**Comment by Zash** [comment]
Late, but yes, it seems to work. Thanks!
